src/plugins/intel_npu/tests/functional/behavior/npu_driver_compiler_adapter/custom_stream.cpp

```diff
@@ -0,0 +1,133 @@
+    }
+
+    std::string generateRandomFileName() {
+        std::stringstream ss;
+        auto now = std::chrono::high_resolution_clock::now();
+        auto seed = now.time_since_epoch().count();
+        std::mt19937 mt_rand(static_cast<unsigned int>(seed));
+        std::uniform_int_distribution<int> dist(0, 15);
+
+        for (unsigned int i = 0; i < 16; ++i) {
+            int random_number = dist(mt_rand);
+            ss << std::hex << random_number;
+        }
+        return ss.str();
+    }
+
+    void SetUp() override {
+        std::tie(target_device, configuration) = this->GetParam();
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+        OVPluginTestBase::SetUp();
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<CompilationParams> obj) {
+        std::string targetDevice;
+        ov::AnyMap configuration;
+        std::tie(targetDevice, configuration) = obj.param;
+        std::replace(targetDevice.begin(), targetDevice.end(), ':', '.');
+
+        std::ostringstream result;
+        result << "targetDevice=" << targetDevice << "_";
+        result << "targetPlatform=" << ov::test::utils::getTestsPlatformFromEnvironmentOr(targetDevice) << "_";
+        if (!configuration.empty()) {
+            for (auto& configItem : configuration) {
+                result << "configItem=" << configItem.first << "_";
+                configItem.second.print(result);
+            }
+        }
+        return result.str();
+    }
+
+    void TearDown() override {
+        if (!configuration.empty()) {
+            utils::PluginCache::get().reset();
+        }
+        APIBaseTest::TearDown();
+    }
+
+protected:
+    ov::AnyMap configuration;
+};
+
+TEST_P(DriverCompilerAdapterCustomStreamTestNPU, TestLargeModel) {
+    auto model = createModelWithLargeSize();
+    IRSerializer irSerializer(model, 11);
+    size_t xmlSize = irSerializer.getXmlSize();
+    size_t weightsSize = irSerializer.getWeightsSize();
+
+    std::vector<uint8_t> xml(xmlSize);
+    std::vector<uint8_t> weights(weightsSize);
+    irSerializer.serializeModelToBuffer(xml.data(), weights.data());
+
+    std::string fileName = generateRandomFileName();
+    std::string xmlFileName = fileName + ".xml";
+    std::string binFileName = fileName + ".bin";
+    {
+        std::ofstream xmlFile(xmlFileName, std::ios::binary);
+        if (xmlFile) {
+            xmlFile.write(reinterpret_cast<const char*>(xml.data()), xmlSize);
+            xmlFile.close();
+        }
+
+        std::ofstream binFile(binFileName, std::ios::binary);
+        if (binFile) {
+            binFile.write(reinterpret_cast<const char*>(weights.data()), weightsSize);
+            binFile.close();
+        }
+    }
+    ov::Core core;
+    EXPECT_NO_THROW(model = core.read_model(xmlFileName));
+    if (std::remove(xmlFileName.c_str()) != 0 || std::remove(binFileName.c_str()) != 0) {
```

Comment:
This should probably be moved to some teardown routine otherwise the file won't be remoted on test failure.

Reply:
Thanks! Moved to teardown now.